pkgs/llvm/default.nix

```diff
@@ -0,0 +1,27 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+{pkgs, ...}: {
+  llvm_cheriot = pkgs.stdenv.mkDerivation rec {
+    name = "llvm-cheriot";
+    src = pkgs.fetchFromGitHub {
+      owner = "CHERIoT-Platform";
+      repo = "llvm-project";
+      rev = "432ef65c96c9f8be8b7802a8bfb9bcf3609b2b79";
+      sha256 = "sha256-ddLnG1olyRYxvKJ0zKEbThq/ZnxP68X32+1ws76WeV4=";
+    };
+    sourceRoot = "${src.name}/llvm";
+    nativeBuildInputs = with pkgs; [cmake ninja lld python3];
+    cmakeFlags = [
```

Comment:
Make sure you're doing a release build, as the debug build can be rather slow. You might still want to enable `LLVM_ENABLE_ASSERTIONS`, though, as this is a fork of mainline LLVM that might have latent bugs we'll encounter.

Reply:
Is release build not the default?

Reply:
I built this derivation locally and inspected the binary with objdump. I think it's indeed not optimised.

Reply:
Good spot, I checked the derivation and you're right the release flag hadn't been added by Nix. I've added the release build type and also assertions while I was at it.

Reply:
I was wrong. Building it again with the additional cmake flags gives the same binary, so the default is indeed Release. I think what I saw probably is the result of outlining.